lib/routing/sfRoute.class.php

```diff
@@ -279,6 +270,45 @@ public function generate($params, $context = [], $absolute = false)
         return $url;
     }
 
+    /**
+     * Generates a consistent cache key for a route
+     */
+    protected function generateCacheKey(array $elements = [], bool $absolute = false): string
+    {
+        $elements = [
+            SharedCacheHelper::ROUTING_NAMESPACE,
+            $this->flattenArrayElements($elements),
+        ];
+
+        if ($absolute) {
+            $elements[] = 'Absolute';
+        }
+
+        return SharedCacheHelper::getNamespace($elements);
+    }
+
+    /**
+     * Recursive method to pack a key => value pair array down into a flat
+     * numerically-indexed array, with a consistent ordering of elements.
+     */
+    protected function flattenArrayElements(array $elements = []): array
+    {
+        $flattened = [];
+        ksort($elements);
```

Comment:
This will make sure we only cache a route once, even if the route arguments are given in a different order in different places. For example:
- `url_for('pack', ['id' => $order_id, 'confirmation' => 1, 'force' => 1])`
- `url_for('pack', ['id' => $order_id, 'force' => 1, 'confirmation' => 1])`

These routes are fundamentally the same, and would result in the same output from the routing engine. To cache them separately because the arguments are provided in a different order would be foolish and wasteful.